src/Http/Http.Abstractions/test/Validation/ValidatableParameterInfoTests.cs

```diff
@@ -285,6 +285,60 @@ public async Task Validate_ExceptionDuringValidation_CapturesExceptionAsError()
         Assert.Equal("Test exception", error.Value.First());
     }
 
+    [Fact]
+    public async Task Validate_WithoutValidationContext_RequiredParameter_AddsErrorAndInitializesValidationContext()
+    {
+        // Arrange
+        var paramInfo = CreateTestParameterInfo(
+            parameterType: typeof(string),
+            name: "testParam",
+            displayName: "Test Parameter",
+            validationAttributes: [new RequiredAttribute()]);
+
+        // Create a ValidateContext without a pre-populated ValidationContext
+        var context = new ValidateContext
+        {
+            ValidationOptions = new TestValidationOptions(new Dictionary<Type, ValidatableTypeInfo>())
+        };
+
+        // Sanity check
+        Assert.Null(context.ValidationContext);
+
+        // Act
+        await paramInfo.ValidateAsync(null, context, default);
+
+        // Assert – a ValidationContext should have been created and the error recorded
+        Assert.NotNull(context.ValidationContext);
```

Comment:
Any concerns with the internally created `ValidationContext` "leaking" outside of the validate call?

Reply:
I don't think so? There isn't any state on the ValidationContext that the caller wouldn't be able to derive anyway.

Ideally, we wouldn't have this type at all and we'd have a single type that somehow unions ValidateContext and ValidationContext.